File: 2017/Day 01/Part 1.py
input = open("input.txt").read().rstrip()

sum = 0

for i in range(len(input) - 1):
    if int(input[i]) == int(input[i + 1]):
        sum += int(input[i])

if int(input[-1]) == int(input[0]):
    sum += int(input[-1])
    
print(sum)

File: 2023/Day 01/Part 1.hs
import Data.Char
import System.IO

headLast :: [Char] -> [Char]
headLast s = [head s, last s]

firstLast :: [[Char]] -> [[Char]]
firstLast = map headLast

main = do
  input <- readFile "input.txt"
  let digitFilter = filter isDigit
  let digits = map digitFilter

  print $ sum $ (map read $ firstLast $ digits $ lines input :: [Integer])